tests/testthat/test-renderer1-knit-print.R

```diff
@@ -1,168 +1,174 @@
-  expect_equal(get_circles(), list(10, 10))
-})
+acontext("knitting multiple animint plots in a single Rmd")
+knitr::knit_meta() #clear knitr 'metadata'
+# Rmd.file <- "~/R/animint/inst/examples/test_knit_print.Rmd" ## Do we need this???
+Rmd.file <- system.file("examples", "test_knit_print.Rmd", 
+                        package = "animint2")
+index.file <- file.path("animint-htmltest", "index.Rmd")
+
+file.copy(Rmd.file, index.file, overwrite=TRUE)
+## https://github.com/rstudio/rmarkdown/issues/587#issuecomment-168437646
+## @yihui says "Do not use the output_dir argument of render()"
+rmarkdown::render(index.file)
+remDr$refresh()
+Sys.sleep(3)
+html <- getHTML()
+
+test_that("knit_print.animint renders five x axis titles", {
+  nodes <- getNodeSet(html, "//text[@class='xtitle']")
+  value.vec <- sapply(nodes, xmlValue)
+  expected.vec <-
+    c("first plot with color legend",
+      "second plot with color legend",
+      "non-interactive plot",
+      "position",
+      "segments")
+  expect_identical(value.vec, expected.vec)
+})
+
+test_that("segments and breakpoints are rendered", {
+  seg.list <- getNodeSet(html, '//g[@class="geom3_segment_signal"]//line')
+  expect_equal(length(seg.list), 6)
+  break.list <- getNodeSet(html, '//g[@class="geom4_vline_signal"]//line')
+  expect_equal(length(break.list), 5)
+})
+
+test_that("svg id property is unique", {
+  svg.list <- getNodeSet(html, "//svg")
+  attr.mat <- sapply(svg.list, xmlAttrs)
+  id.counts <- table(attr.mat["id",])
+  expect_true(all(id.counts==1))
+})
+
+all.list <- getNodeSet(html, "//*")
+id.na.vec <- sapply(all.list, function(e){
+  attr.vec.or.null <- xmlAttrs(e)
+  if("id" %in% names(attr.vec.or.null)){
+    attr.vec.or.null[["id"]]
+  }else{
+    NA
+  }
+})
+## In HTML, all values are case-insensitive
+## http://www.w3schools.com/tags/att_global_id.asp
+lower.id.vec <- tolower(id.na.vec)
+id.counts <- table(lower.id.vec)
+(not.unique <- id.counts[1 < id.counts])
+test_that("id property is unique over entire page", {
+  expect_equal(length(not.unique), 0)
+})
+
+test_that("id must contain at least one character", {
+  expect_true(all(0 < nchar(names(id.counts))))
+})
+
+test_that("id must not contain any space characters", {
+  expect_false(any(grepl(" ", names(id.counts))))
+})
+
+## function to extract all circles from an HTML page
+get_circles <- function(html=getHTML()) {
+  plot.names <- c("plot1top", "plot1bottom")
+  count.vec <- c()
+  for(i in seq_along(plot.names)){
+    xpath <- sprintf("//div[@id='%s']//circle[@class='geom']", plot.names[[i]])
+    circle.list <- getNodeSet(html, xpath)
+    count.vec[[i]] <- length(circle.list)
+  }
+  count.vec
+}
+
+get_elements <- function(id){
+  
+  top_widget_show_hide<- runtime_evaluate_helper(id=id,class_name='show_hide_selector_widgets',list_num=0,return.value=TRUE)
+  bottom_widget_show_hide<- runtime_evaluate_helper(id=id,class_name='show_hide_selector_widgets',list_num=0,return.value=TRUE)
```

Comment:
before this PR we have
```r
  a <- tr.list[[1]]
  b <- tr.list[[2]]
```
so I believe on this line you should have list_num=1, to get the second element.
Currently this line and the previous line both have list_num=0, which gets the first element, right?

Reply:
I am not sure about that. but list number 0 is working.

Reply:
can you please investigate? are both lines returning the same value? if so, can you please explain why the test passes? I expect the test should fail if there is a bug/substitution (0 instead of 1) in the test code.
should we update the test/expectation here to be more robust?